lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1132,24 +1145,33 @@ def sanitize_cdrom_params(self):
                 if cdrom_spec['controller_type'] == 'ide' and \
                         (cdrom_spec.get('controller_number') not in [0, 1] or cdrom_spec.get('unit_number') not in [0, 1]):
                     self.module.fail_json(msg="Invalid cdrom.controller_number: %s or cdrom.unit_number: %s, valid"
-                                              " values are 0 or 1 for IDE controller." % (cdrom_spec.get('controller_number'), cdrom_spec.get('unit_number')))
+                                              " values are 0 or 1 for IDE controller."
+                                              % (cdrom_spec.get('controller_number'), cdrom_spec.get('unit_number')))
 
                 if cdrom_spec['controller_type'] == 'sata' and \
                         (cdrom_spec.get('controller_number') not in range(0, 4) or cdrom_spec.get('unit_number') not in range(0, 30)):
                     self.module.fail_json(msg="Invalid cdrom.controller_number: %s or cdrom.unit_number: %s,"
                                               " valid controller_number value is 0-3, valid unit_number is 0-29"
-                                              " for SATA controller." % (cdrom_spec.get('controller_number'), cdrom_spec.get('unit_number')))
+                                              " for SATA controller." % (cdrom_spec.get('controller_number'),
+                                                                         cdrom_spec.get('unit_number')))
 
                 ctl_exist = False
-                for exist_spec in cdroms.get(cdrom_spec['controller_type']):
-                    if exist_spec['num'] == cdrom_spec['controller_number']:
+                for exist_spec in cdrom_specs:
```

Comment:
cdrom_specs[] is not initialized before this. am I missing anything here?

Reply:
It is initialized at the beginning of this function. Thanks.

Reply:
No! Sorry I used wrong word initialized.
I meant no value is assigned to cdrom_specs[]

Reply:
Yeah, for the first time it is empty, then the configured cdrom info will be added to it, `cdrom_specs.append({'ctl_num': cdrom_spec['controller_number'], 'ctl_type': cdrom_spec['controller_type'], 'cdroms': [cdrom_spec]})`